src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(operand) {
+      calculator.result += operand;
+    },
+
+    subtract(operand) {
+      calculator.result -= operand;
+    },
+
+    multiply(operand) {
+      calculator.result *= operand;
+    },
+
+    divide(operand) {
+      calculator.result /= operand;
+    },
+
+    reset() {
+      calculator.result = 0;
+
+      return this;
+    },
+
+    operate(callback, operand) {
+      callback(operand);
```

Comment:
The 'callback' function might potentially change the context, so it's safer to bind the current context ('this') to it. Use 'callback.call(this, operand)' instead of 'callback(operand)'.

Reply:
ok, thanks